lib/use_cases/use_case_factory.rb

```diff
@@ -0,0 +1,13 @@
+class UseCaseFactory
+  def update_rent_charges
+    UpdateRentCharges.new(fixed_data_gateway: fixed_data_gateway, rent_charges_gateway: rent_charges_gateway)
+  end
+
+  def fixed_data_gateway
```

Comment:
Would recommend making the `*_gateway` methods private, as controllers shouldn't have access to them directly. Only the use cases need them.